### PR TITLE
[webui] handle obscureText setting for single line TextField

### PR DESCRIPTION
**Pull request by tauu**
Though setting obscureText to true for TextField / TextFormField widgets will obscure the rendered  text content, it is still fully visible in the suggestion / autocomplete bar of soft keyboards (and touchbars of MacBooks). This PR changes this behaviour by setting the type of the input element corresponding to the TextField to 'password'. Unfortunately this works only for single line TextFields.

### CODE REVIEW

**Comment by nturgut** [comment]
Thanks for sending out this PR! 

Looks like there are some issues:
1) There is a trailing white space. In my opinion the easiest solution to this is to configure the IDE to remove the trailing white spaces. In VS Code I use these lines in my settings JSON:
  "files.trimTrailingWhitespace": true,
  "editor.trimAutoWhitespace": true,

2) The tests are failing. We have a testing tool now. You can use it for running all the tests. Here are the instructions: https://github.com/flutter/engine/blob/master/lib/web_ui/dev/README.md I also made some changes in that file today. It might be best to fetch+rebase before running the tests.

Cheers,

Nurhan

**Comment by tauu** [comment]
Thanks for the tips regarding the VSCode setting as well as the new testing tool (very nice improvement :-) ).

The failing test were fixed by rebasing to the current master branch.

File: lib/web_ui/lib/src/engine/text_editing.dart [hunk 1]
```diff
@@ -330,6 +330,11 @@ class TextEditingElement {
     switch (inputConfig.inputType) {
       case InputType.text:
         domElement = owner.createInputElement();
+        // Setting the type to password seems to be only option to prevent the obscured text
```
Comment (nturgut) [inline lib/web_ui/lib/src/engine/text_editing.dart]: Can you add a unit test for testing this configuration?

In test_editing_test.dart (line 20-35) there are 2 input configurations now. You can also add a third one with type text and obscuretext=true. Later use it in a test case.

File: lib/web_ui/lib/src/engine/text_editing.dart [hunk 2]
```diff
@@ -330,6 +330,11 @@ class TextEditingElement {
...
+        // from being displayed as an suggestion / autocomplete option in mobile devices.
```
Comment (nturgut) [inline lib/web_ui/lib/src/engine/text_editing.dart]: nit: small typo, a suggestion.

**Comment by mdebbar** [comment]
@tauu are you still planning to address the comments and submit the PR?

**Comment by cbracken** [comment]
Thanks for your contribution. Since we haven't seen a reply, we're going to reluctantly close this patch for now, but if you get time to address the comments, we'd be more than happy to take a look at a new patch. If you do send a new patch, it would be very useful to reference this one in the description in order to help reviewers see the previous context.

**Pull request closed without merging**